test/client/fixtures/sandbox/node/window-test.js

```diff
@@ -11,6 +11,31 @@ test('window.onerror setter/getter', function () {
     strictEqual(getProperty(window, 'onerror'), handler);
 });
 
+module('timers');
+
+asyncTest('setTimeout script argument processing', function () {
```

Comment:
If we change the strategy, we still have to rewrite these tests. We can not test this particular case as a black box. To do this, we need functional tests.